src/main/kotlin/com/google/coroutinestacks/CoroutineTraceForestBuilder.kt

```diff
@@ -121,11 +106,14 @@ private fun SuspendContextImpl.createCoroutineTraceForest(
     )
 }
 
-fun createCoroutineTraces(rootValue: Node): List<CoroutineTrace?> {
+fun createCoroutineTraces(rootValue: Node): Pair<MutableList<CoroutineTrace>, MutableMap<Int, MutableList<Int>>> {
```

Comment:
Why do we need to carry around coroutine trace children separately? Let's add them as a field to the CoroutineTrace class:
```
data class CoroutineTrace(
    val stackFrameItems: MutableList<CoroutineStackFrameItem?>,
    val runningCount: Int,
    val suspendedCount: Int,
    val coroutinesActiveLabel: String,
    val children: MutableList<CoroutineTrace>
)
```